t/transactions-spec.t

```diff
@@ -0,0 +1,667 @@
+#
+#  Copyright 2015 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+use JSON::MaybeXS qw( is_bool decode_json );
+use Path::Tiny 0.054; # basename with suffix
+use Test::More 0.96;
+use Test::Deep;
+use Math::BigInt;
+use Storable qw( dclone );
+
+use utf8;
+
+use MongoDB;
+use MongoDB::_Types qw/
+    to_IxHash
+/;
+use MongoDB::Error;
+
+use lib "t/lib";
+
+use if $ENV{MONGOVERBOSE}, qw/Log::Any::Adapter Stderr/;
+
+use MongoDBTest qw/
+    build_client
+    get_test_db
+    server_version
+    server_type
+    clear_testdbs
+    get_unique_collection
+    skip_unless_mongod
+    skip_unless_failpoints_available
+/;
+
+# TODO Keep not getting hosts???? skip_unless_mongod();
+# TODO skip_unless_failpoints_available();
+
+my @events;
+
+# TODO strip all Dwarns
+#use Devel::Dwarn;
+
+sub clear_events { @events = () }
+sub event_count { scalar @events }
+# Must use dclone, as was causing action at a distance for binc on txn number
+sub event_cb { push @events, dclone $_[0] }#; Dwarn $_[0] }
+
+my $conn           = build_client();
+my $server_version = server_version($conn);
+my $server_type    = server_type($conn);
+
+plan skip_all => "Requires MongoDB 4.0"
```

Comment:
For the 3.6 tests, I added a test helper function for "skip_unless_sessions".  Please make a similar one for "skip_unless _transactions" here.